crates/ide_completion/src/render/function.rs

```diff
@@ -562,6 +562,78 @@ fn qux(Foo { bar }: Foo) {
 fn main() {
   qux(${1:foo})$0
 }
+"#,
+        );
+    }
+
+    #[test]
+    fn complete_fn_param() {
+        // has mut kw
+        check_edit(
+            "mut bar",
+            r#"
+fn f(foo: (), mut bar: u32) {}
+fn g(foo: (), mut ba$0)
+"#,
+            r#"
+fn f(foo: (), mut bar: u32) {}
+fn g(foo: (), mut bar: u32)
+"#,
+        );
+
+        // has type param
+        check_edit(
+            "mut bar",
+            r#"
+fn g(foo: (), mut ba$0: u32)
+fn f(foo: (), mut bar: u32) {}
+"#,
+            r#"
+fn g(foo: (), mut bar: u32)
+fn f(foo: (), mut bar: u32) {}
+"#,
+        );
+    }
+
+    #[test]
+    fn complete_fn_mut_param_add_comma() {
+        // add leading and trailing comma
+        check_edit(
+            "mut bar",
+            r#"
+fn f(foo: (), mut bar: u32) {}
+fn g(foo: ()mut ba$0 baz: ())
+"#,
+            r#"
+fn f(foo: (), mut bar: u32) {}
+fn g(foo: (), mut bar: u32, baz: ())
+"#,
+        );
+    }
+
+    #[test]
+    fn complete_fn_mut_param_has_attribute() {
+        check_edit(
+            "mut bar",
+            r#"
+fn f(foo: (), #[baz = "qux"] mut bar: u32) {}
+fn g(foo: (), mut ba$0)
```

Comment:
Tests pass for both completion listing and edit, but doesn't work on my machine, not sure what's wrong.

Reply:
This might be the same bug that I fixed in https://github.com/rust-lang/rust-analyzer/pull/11967 - basically, the completion's `lookup_by` must start with the characters in its range, otherwise the editor will filter out the completion

Reply:
If I understand correctly, 8f8f20fda5936faff04f9c5c3c8880bd4572900f should fix this? At least it became working in my environment.